### PR TITLE
Apply page

### PR DESCRIPTION
**Pull request by rymaju**
Added:

-  Apply page using card layout. The card layout was chosen over the tabs layout (still included just in case we want to use that work) because it looks better on mobile.

- Apply page cards are their own component, they are heavily styled after the Uber cards layout. Card elevation is set to 0 (there is no border) and the has a cool underline transition animation

- Apply page cards have a delayed fade in transition, i think it looks cool and it loads well on mobile

- One "role" page, for a Developer, its best to abstract the common elements out into a template component just for role pages, but since we only have one page right now and the design isnt finalized I left it as one .tsx file for now. The reason we need a separate page for each role is for SEO purposes (it gets its own Helmet tag and its place on our sitemap)

image

image

image

image

image

image

### CODE REVIEW

File: public/index.html
Comment (jackblanc) [inline public/index.html:34]: What's this doing?

File: src/App.tsx
Comment (jackblanc) [inline src/App.tsx]: ⛏️ (nitpick - small comment)
these extra lines could be removed :)

File: src/App.tsx
Comment (jackblanc) [inline src/App.tsx:7]: Developer seems a little too vauge to me, being in the components directory. 

I think a clean solution is organizing the files into subdirectories by page, for anything specific to one of the 7 pages, and keep general components in /components.

Comment (jackblanc) [reply src/App.tsx:7]: @54skyxenon hold off on making any changes related to file structure (I left some comments on your PR) until we agree on a pattern to follow. should discuss on monday as a group.

File: src/components/ApplyGrid.tsx
Comment (jackblanc) [inline src/components/ApplyGrid.tsx]: This component could use a more descriptive name

I think it'll be worth creating a generalized component that takes content as props, but you don't have to in this PR.

File: src/components/ApplyTabs.tsx
Comment (jackblanc) [inline src/components/ApplyTabs.tsx]: Can you add a link please

File: src/components/ApplyCard.tsx
Comment (jackblanc) [inline src/components/ApplyCard.tsx]: Design + marketing might want us to use the IBM Plex Sans for everything

File: src/components/ApplyCard.tsx
Comment (jackblanc) [inline src/components/ApplyCard.tsx:34]: this is really cool 🔥

File: src/components/ApplyCard.tsx
Comment (jackblanc) [inline src/components/ApplyCard.tsx]: add `readonly title: string;`, honestly haven't seen it catch anything yet but it's a best practice afaik

File: src/components/ApplyCard.tsx
Comment (jackblanc) [inline src/components/ApplyCard.tsx]: Can you also add doc comments for this? `/** */`

Side note: look into Storybook.js. It might be worth installing asap, I like it a lot.

File: src/components/ApplyCard.tsx
Comment (jackblanc) [inline src/components/ApplyCard.tsx]: Can you destructure the props here? 
```suggestion
const ApplyCard: React.FC<RoleProps> = ({
  title,
  children,
  to,
  enterTime,
}) => {
```

Comment (jackblanc) [reply src/components/ApplyCard.tsx]: (don't use github to commit this IMO, just do locally)

File: src/components/ApplyCard.tsx
Comment (jackblanc) [inline src/components/ApplyCard.tsx]: Dumb question: do we have to specify the type of children?

File: src/components/ApplyCard.tsx
Comment (jackblanc) [inline src/components/ApplyCard.tsx]: Can we get around using inline styling here?

File: src/components/ApplyTabs.tsx
Comment (jackblanc) [inline src/components/ApplyTabs.tsx]: Could you add a comment explaining why we need this? 

Also, why is event listed if it's unused?

File: src/components/ApplyTabs.tsx
Comment (jackblanc) [inline src/components/ApplyTabs.tsx]: confused what these are doing?

File: src/components/NavBar.tsx
Comment (jackblanc) [inline src/components/NavBar.tsx:25]: ⛏️  remove logo? honestly not entirely sure what alt does

File: src/theme.tsx
Comment (jackblanc) [inline src/theme.tsx:5]: What's this for?

**Review by jackblanc — COMMENTED** [review]
Great work here, the new pages look amazing! I left a lot of comments, as this is a pretty beefy PR. 

We should talk Monday about the file organization stuff, and review some of the best practices used here as a group. 

Excited to see this coming together so well!

File: public/index.html
Comment (rymaju) [reply public/index.html:34]: Loading in fonts for light, regular, medium, bold + italics for each. We can always remove what we dont use but I decided to load everything for now just for ease of use

File: src/App.tsx
Comment (rymaju) [reply src/App.tsx:7]: Hmm, yeah I think we could have a "Pages" folder, which contains a folder for each page and under each folder is both the page .tsx file and its components: i.e. src/pages/apply/Apply.tsx, src/pages/apply/ApplyCard.tsx, etc.

File: src/components/ApplyCard.tsx
Comment (rymaju) [reply src/components/ApplyCard.tsx]: This was an issue with custom styled components + the "component" prop. I dont think theres a paticularly elegant way to get this to work with styled buttons. I can change this to a css class rather an inline though.

File: src/components/ApplyTabs.tsx
Comment (rymaju) [reply src/components/ApplyTabs.tsx]: Honestly, ill just delete the ApplyTabs component. We arent going to use it anyway

File: src/components/NavBar.tsx
Comment (rymaju) [reply src/components/NavBar.tsx:25]: In the case an image load fails it will load the alt tag as text. Also screen readers will read the alt tag which is why I have the "logo" bit.

File: src/theme.tsx
Comment (rymaju) [reply src/theme.tsx:5]: I think this was leftover from me playing around with the theming of the site, I can un-uncomment this

Comment (rymaju) [reply src/theme.tsx:5]: Actually I wont, I'm going to leave it like this because we dont know what our color scheme should be yet and I think its best to leave it to MUI defaults. Currently the master branch has some theme styling that I put just to show how styling *could* be done. But honestly, looks kind of bad.

File: src/components/Apply.tsx
Comment (jackblanc) [inline src/components/Apply.tsx]: should these commented styles be removed?

File: src/components/Apply.tsx
Comment (jackblanc) [inline src/components/Apply.tsx:5]: unnecessary line here

**Review by jackblanc — COMMENTED** [review]
It looks like there are a few open comments still

File: src/components/Apply.tsx
Comment (rymaju) [reply src/components/Apply.tsx]: This is from when I was playing around with gradients, I'll just remove it for now

File: src/components/Apply.tsx
Comment (rymaju) [reply src/components/Apply.tsx:5]: Not exactly sure what you mean here?

Comment (jackblanc) [reply src/components/Apply.tsx:5]: sorry that wasn't clear, we don't need the extra line between the import statements

File: src/components/Apply.tsx
Comment (jackblanc) [reply src/components/Apply.tsx]: Great, yeah we just don't need (code) commented in master i'd say

**Review by jackblanc — APPROVED** [review]
If you wanna add comments to `RoleProps` that'd be great but not blocking

**Pull request merged**
